.github/workflows/build-publish.yaml

```diff
@@ -0,0 +1,81 @@
+name: Build Jellyfin-Kodi
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        py_version: [ 'py2', 'py3' ]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python 3.x
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install pyyaml
+
+      - name: Create ${{ matrix.py_version }} addon.xml
+        run: python .config/generate_xml.py ${{ matrix.py_version }}
+
+      - name: Publish Build Atrifact
+        uses: actions/upload-artifact@v2
+        with:
+          retention-days: 14
+          name: ${{ matrix.py_version }}-build-artifact
+          path: |
+            **/*
+            !.ci/*
+            !.config/*
+            !.git/**/*
+            !.github/*
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ startsWith(github.ref, 'refs/tags/') && github.repository == 'jellyfin/jellyfin-kodi' }}
+    strategy:
+      matrix:
+        py_version: [ 'py2', 'py3' ]
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: ${{ matrix.py_version }}-build-artifact
+          path: plugin.video.jellyfin-${{ matrix.py_version }}
+
+      - name: Create release Zip
+        run: zip -rq plugin.video.jellyfin-${{ matrix.py_version }}.zip plugin.video.jellyfin-${{ matrix.py_version }}
```

Comment:
This is actually something I've been thinking about changing for a while.  I may have mentioned in in chat, but I don't remember.  In JellyCon I made a build script that combines the zipping with the `addon.xml` generation into one.  If we're already changing the build system around, it might make sense to port that over here now.  I'm open to suggestions about it.

Refs:

* https://github.com/jellyfin/jellycon/blob/master/build.py
* https://github.com/jellyfin/jellycon/blob/master/.ci/build.yml#L28-L29

Reply:
I will have a look!

(btw. woudn't https://github.com/jellyfin/jellyfin-kodi/pull/405 be a ref too?)

Reply:
It would, yes.  It stalled out because we couldn't agree on the dependencies, and then I ended up writing my own for JellyCon that combined our approaches and #405 never got picked back up afterwards.